mm2src/db_common/src/sql_query.rs

```diff
@@ -316,6 +316,23 @@ impl<'a> SqlQuery<'a> {
             .field(format!("ROW_NUMBER() OVER (ORDER BY {}) AS {}", order_by, alias));
         Ok(self)
     }
+
+    /// Count all rows
+    pub fn count_all(&mut self) -> SqlResult<&mut Self> {
+        self.sql_builder.count("*");
+        Ok(self)
+    }
+
+    /// Select from union tables
+    pub fn select_from_union_alias(conn: &'a Connection, union_sql: &str, alias: &'static str) -> SqlResult<Self> {
+        validate_table_name(alias)?;
+        Ok(SqlQuery {
+            conn,
+            sql_builder: SqlBuilder::select_from(format!("({}) AS {}", union_sql, alias)),
```

Comment:
Vulnerable agains't sql injection attacks

Reply:
Sorry how this can be manipulated if I validate table and alias names?

And for instance according this function
```rust
fn get_nft_list_builder_preimage(conn: &Connection, chains: Vec<Chain>) -> MmResult<SqlQuery, SqlError> {
    let union_sql_strings = chains
        .iter()
        .map(|chain| {
            let table_name = nft_list_table_name(chain);
            validate_table_name(&table_name)?;
            let sql_builder = SqlQuery::select_from(conn, table_name.as_str())?;
            let sql_string = sql_builder.sql()?.trim_end_matches(';').to_string();
            Ok(sql_string)
        })
        .collect::<MmResult<Vec<_>, SqlError>>()?;
    let union_sql = union_sql_strings.join(" UNION ALL ");
    let mut final_sql_builder = SqlQuery::select_from_union_alias(conn, union_sql.as_str(), "nft_list")?;
    final_sql_builder.order_desc("nft_list.block_number")?;
    drop_mutability!(final_sql_builder);
    Ok(final_sql_builder)
}
```
chain name must be matched with enum variant

well seems like I can use this [functon](https://github.com/KomodoPlatform/atomicDEX-API/blob/dev/mm2src/db_common/src/sql_query.rs#L35)

Reply:
I don't know how exactly that function works, but you put 2 values inside of the SQL without prepared parameters, in case if any of them include malicious query, it will work.

Reply:
Ok I rechecked, I cant use `pub fn select_from_alias`, bcz I have union of tables, and validate_table_name(table)?; will fail.
Then I will build SqlQuery by myself right in `builder_preimage` functions.

Reply:
Can you simply build prepared statement and use it instead of `SqlQuery` ? AFAIK `SqlQuery` was created manually by Sergey which was all about string building(that what makes it vulnerable to injections).

Reply:
added prepared statements into `get_nft_list` and `get_tx_history`